website/docs/r/network_connection_monitor.html.markdown

```diff
@@ -74,41 +157,131 @@ The following arguments are supported:
+---
+
+A `item` block supports the following:
 
-* `address` - (Optional) The address of the connection monitor destination (IP or domain name). Conflicts with `destination.0.virtual_machine_id`
+* `type` - (Optional) The type of item included in the filter. Possible value is `AgentAddress`. Defaults to `AgentAddress`.
 
-* `virtual_machine_id` - (Optional) The ID of the virtual machine used as the destination by connection monitor. Conflicts with `destination.0.address`
+* `address` - (Optional) The address of the filter item.
 
 ---
 
-A `source` block supports the following:
+A `test_configuration` block supports the following:
+
+* `name` - (Required) The name of the connection monitor test configuration.
+
+* `protocol` - (Required) The protocol to use in test evaluation. Possible values are `Tcp`, `Http` and `Icmp`.
+
+* `test_frequency_sec` - (Optional) The frequency of test evaluation, in seconds. Defaults to `60`.
+
+* `http_configuration` - (Optional) A `http_configuration` block as defined below.
+
+* `icmp_configuration` - (Optional) A `icmp_configuration` block as defined below.
+
+* `preferred_ip_version` - (Optional) The preferred IP version to use in test evaluation. Possible values are `IPv4` and `IPv6`. 
+
+* `success_threshold` - (Optional) A `success_threshold` block as defined below.
+
+* `tcp_configuration` - (Optional) A `tcp_configuration` block as defined below.
+
+---
+
+A `http_configuration` block supports the following:
+
+* `method` - (Optional) The HTTP method to use. Possible values are `Get` and `Post`. Defaults to `Get`.
+
+* `port` - (Optional) The port to connect to.
+
+* `path` - (Optional) The path component of the URI. For instance, "/dir1/dir2".
+
+* `prefer_https` - (Optional) Will https be preferred over http in cases where the choice is not explicit? Defaults to `false`.
+
+* `request_header` - (Optional) A `request_header` block as defined below.
+
+* `valid_status_code_ranges` - (Optional) The http status codes to consider successful. For instance, "2xx, 301-304, 418".
+
+---
+
+A `request_header` block supports the following:
+
+* `name` - (Required) The name in HTTP header.
+
+* `value` - (Required) The value in HTTP header.
+
+---
+
+A `icmp_configuration` block supports the following:
+
+* `disable_trace_route` - (Optional) Will path evaluation with trace route be disabled? Defaults to `false`.
+
+---
+
+A `success_threshold` block supports the following:
+
+* `checks_failed_percent` - (Optional) The maximum percentage of failed checks permitted for a test to evaluate as successful.
+
+* `round_trip_time_ms` - (Optional) The maximum round-trip time in milliseconds permitted for a test to evaluate as successful.
+
+---
+
+A `tcp_configuration` block supports the following:
+
+* `port` - (Required) The port to connect to.
+
+* `disable_trace_route` - (Optional) Should path evaluation with trace route be disabled? Defaults to `false`.
```

Comment:
we always end with enabled/disabled
```suggestion
* `trace_route_disabled` - (Optional) Should path evaluation with trace route be disabled? Defaults to `false`.
```

or even change it to `trace_route_enabled`

Reply:
updated